platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/PreResolvedResolvableArtifact.java

```diff
@@ -34,26 +32,25 @@
 import javax.annotation.Nullable;
 import java.io.File;
 
-public class PreResolvedResolvableArtifact implements ResolvableArtifact, ResolvedArtifact {
+public class PreResolvedResolvableArtifact implements ResolvableArtifact {
```

Comment:
💭 If it's been pre-resolved, is it really still resolvable?  If so, doesn't that make it re-resolvable?  

Even with `ResolvedArtifact` already taken to mean something different, this name is a little goofy.

Reply:
Yeah I think this probably a legacy name from when before we have a `PreResolved` version. Before CC and artifact transforms existed, all `ResolvableArtifacts` were resolvable and didn't know their file beforehand. 

I think this could use a better name, but this also seems like a pretty core artifact resolution type. I'd prefer not to change the name here